packages/lodestar/src/api/impl/beacon/state/state.ts

```diff
@@ -1,26 +1,159 @@
+    const state = await this.getState(stateId);
+    if (!state) {
+      return null;
+    }
+    return {
+      currentJustified: state.currentJustifiedCheckpoint,
+      previousJustified: state.previousJustifiedCheckpoint,
+      finalized: state.finalizedCheckpoint,
+    };
+  }
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  public async getStateValidators(stateId: StateId, filters?: IValidatorFilters): Promise<ValidatorResponse[]> {
+    const state = await resolveStateId(this.config, this.db, this.forkChoice, stateId);
+    if (!state) {
+      throw new StateNotFound();
+    }
+    //TODO: implement filters
+    return readOnlyMap(state.state.validators, (v, index) => toValidatorResponse(index, v));
+  }
+
+  public async getStateValidator(
+    stateId: StateId,
+    validatorId: ValidatorIndex | Root
+  ): Promise<ValidatorResponse | null> {
+    const state = await resolveStateId(this.config, this.db, this.forkChoice, stateId);
+    if (!state) {
+      throw new StateNotFound();
+    }
+    let validatorIndex: ValidatorIndex | undefined;
+    if (typeof validatorId === "number") {
+      if (state.state.validators.length > validatorId) {
+        validatorIndex = validatorId;
+      }
+    } else {
+      validatorIndex = (await this.chain.getHeadEpochContext()).pubkey2index.get(validatorId) ?? undefined;
+      //validator added later than given stateId
+      if (validatorIndex && validatorIndex >= state.state.validators.length) {
+        validatorIndex = undefined;
+      }
+    }
+    if (!validatorIndex) {
+      throw new ApiError(404, "Validator not found");
+    }
+    return toValidatorResponse(validatorIndex, state.state.validators[validatorIndex]);
+  }
+  public async getStateValidatorBalances(
+    stateId: StateId,
+    indices?: (ValidatorIndex | Root)[]
+  ): Promise<ValidatorBalance[]> {
+    const state = await resolveStateId(this.config, this.db, this.forkChoice, stateId);
+    if (!state) {
+      throw new StateNotFound();
+    }
+    if (indices) {
+      const epochCtx = await this.chain.getHeadEpochContext();
+      return indices
+        .map((id) => {
+          if (typeof id === "number") {
+            if (state.state.validators.length <= id) {
+              return null;
+            }
+            return {
+              index: id,
+              balance: state.state.balances[id],
+            };
+          } else {
+            const index = epochCtx.pubkey2index.get(id) ?? undefined;
+            return index && index <= state.state.validators.length
+              ? {index, balance: state.state.balances[index]}
+              : null;
+          }
+        })
+        .filter(notNullish);
+    }
+    return readOnlyMap(state.state.validators, (v, index) => {
+      return {
+        index,
+        balance: state.state.balances[index],
+      };
+    });
+  }
+  public async getStateCommittees(stateId: StateId, filters?: ICommitteesFilters): Promise<BeaconCommitteeResponse[]> {
```

Comment:
Could you add more space (newlines) between code blocks? It helps readability a lot

Reply:
I hoped prettier would take care of that :disappointed: